buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/TestClusterCleanupOnShutdown.java

```diff
@@ -19,16 +19,12 @@
 
     private Set<ElasticsearchCluster> clustersToWatch = new HashSet<>();
 
-    public void watch(Collection<ElasticsearchCluster> cluster) {
-        synchronized (clustersToWatch) {
-            clustersToWatch.addAll(clustersToWatch);
-        }
+    public synchronized void watch(Collection<ElasticsearchCluster> cluster) {
+        clustersToWatch.addAll(clustersToWatch);
     }
 
-    public void unWatch(Collection<ElasticsearchCluster> cluster) {
-        synchronized (clustersToWatch) {
-            clustersToWatch.removeAll(clustersToWatch);
-        }
+    public synchronized void unWatch(Collection<ElasticsearchCluster> cluster) {
+        clustersToWatch.removeAll(clustersToWatch);
```

Comment:
Shouldn't this be `clustersToWatch.removeAll(cluster)`? (note, the name `cluster` shoudl really be plural since it is a collection?

Reply:
Good catch. Looks like we were never actually registering this clusters to be watched.